### ISSUE TITLE
Better shadows

### ISSUE THREAD

**Issue by zezic** [matinlotfali/KDE-Rounded-Corners#191]
![better-shadows](https://github.com/matinlotfali/KDE-Rounded-Corners/assets/1945127/639083b1-b191-42aa-b675-f39697f499b8)
Related to #153

Dear @matinlotfali, thank you for this amazing effect!
Here is the explained idea for getting more natural shadows. I tried it partially by modifying the shaders and my experiments looks promising, but I'm not so good at shaders (I got good beautiful shadows, but broke corners geometry) so it would be cool if you could make the changes.

Basically, we want to draw the shadow like it's dropped by rectangle which is smaller by X (red arrow) units on all its sides and shifted down by Y (blue arrow) units. It would be cool to get these (X, Y) parameters adjustable in UI.

*P.S. I know, it's still far from ideal natural shadow, but at least it's a good approximation.*

**Comment by matinlotfali** [comment]
@zezic  From your image, if the outline radius is `r`, can you tell me how long are the red and green arrows? Are they both `r`?

**Comment by 0xarch** [comment]
@matinlotfali 
![图片](https://github.com/matinlotfali/KDE-Rounded-Corners/assets/89088411/31b680af-d064-40dc-a303-8b5c709c84b7)
Here's my implementation of this issue, the length of `u`,`v` is `1`, _r_ of `OutlineRadius` is `sqrt(pi)`, so I think it's `r*r/pi`. However it should be _customizable_

**Comment by 0xarch** [comment]
![图片](https://github.com/matinlotfali/KDE-Rounded-Corners/assets/89088411/80d3815d-bed5-45c5-8f45-413df32bda89)
Here's some _other_ thing I want to add

**Comment by matinlotfali** [comment]
@zezic @0xarch 
Is #244 what you had in mind?
